### PR TITLE
feat: add identities list endpoint and support params for it

### PR DESCRIPTION
**Pull request by JohnyArkade**
To retrieve a list of identities, use the `list` method:
We could also add params to list, like `?filter[handle]=handle&include=profile`

```javascript
omneoClient.identities.list()
    .then((identities) => {
        console.log(identities);
    })
    .catch((error) => {
        console.error(error);
    });
```

### CODE REVIEW

**Review by oliverarkade — CHANGES_REQUESTED** [review]
needs tests

File: src/tests/integration/omneo/identities/list-identity.test.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    const sdkIdentityList = await omneo.identities.list()
+    sdkIdentityList.forEach((identity) => {
+      expect(identity).toHaveProperty('id')
+      expect(identity).toHaveProperty('handle')
+      expect(identity).toHaveProperty('identifier')
+      expect(identity).toHaveProperty('profile_id')
+      expect(identity).toHaveProperty('is_active')
+    })
+  })
+  test('SDK can get identities.', async () => {
```
Comment (oliverarkade) [inline src/tests/integration/omneo/identities/list-identity.test.ts:22]: SDK can get identities with filters

File: src/tests/integration/omneo/identities/list-identity.test.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    sdkIdentityList.forEach((identity) => {
+      expect(identity).toHaveProperty('id')
+      expect(identity).toHaveProperty('handle', 'apple_device_id')
+      expect(identity).toHaveProperty('identifier')
+      expect(identity).toHaveProperty('profile_id')
+      expect(identity).toHaveProperty('is_active')
+      expect(identity).toHaveProperty('profile')
+    })
+  })
+  test('SDK can get identities.', async () => {
```
Comment (oliverarkade) [inline src/tests/integration/omneo/identities/list-identity.test.ts:37]: SDK can get identities with pagination

**Review by oliverarkade — CHANGES_REQUESTED** [review]

**Review by oliverarkade — APPROVED** [review]

File: src/tests/integration/omneo/identities/list-identity.test.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      handle: identityHandleName,
+      identifier: randomString(11)
+    }
+    const { id, handle, profile_id: profileID } = await omneo.profiles.createIdentity(decodedProfileID, payload)
+    expect(profileID).toBe(decodedProfileID)
+    expect(handle).toBe(identityHandleName)
+    CREATED_IDENTITIES.push(id)
+
+    const params: RequestParams = {
+      'filter[handle]': 'apple_device_id',
```
Comment (oliverarkade) [inline src/tests/integration/omneo/identities/list-identity.test.ts:63]: should this filter identityHandleName

File: src/tests/integration/omneo/identities/list-identity.test.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      include: 'profile',
+      withPagination: true
+    }
+    const { links, meta }: IdentityResponse = await omneo.identities.list(params)
```
Comment (oliverarkade) [inline src/tests/integration/omneo/identities/list-identity.test.ts:67]: check that every member of the returned array has a handle `identityHandle`
This proves the filter works, and is not just returning anything

**Review by oliverarkade — CHANGES_REQUESTED** [review]

File: src/tests/integration/omneo/identities/list-identity.test.ts (hunk 3 above)
Comment (JohnyArkade) [reply src/tests/integration/omneo/identities/list-identity.test.ts:63]: Yes, changed.

**Review by oliverarkade — APPROVED** [review]

File: src/tests/integration/omneo/identities/list-identity.test.ts (hunk 4 above)
Comment (JohnyArkade) [reply src/tests/integration/omneo/identities/list-identity.test.ts]: fixed, great tips.

**Pull request merged**